### DISCUSSION TITLE
Unix port (custom compile) cannot find panic_handler.mpy.

### DISCUSSION THREAD

**Discussion by RSC-Games** [micropython/micropython#10055] [Using MicroPython] [answered]
I have a stable codebase that works fully on an ESP32. I decided to take this same code and make it run on the Unix port. After a lot of work, I managed to get the code to (mostly) boot, but my code crashes with this:

```
D (571) init: Skipping card init and mount process.
W: append hook.
D: fix_path: /mnt/c/Users/robocoder/Downloads/sp_emu/sys/bin
V: Path: ['/mnt/c/Users/robocoder/Downloads/sp_emu/bin', '.frozen', '/home/robocoder/.micropython/lib', '/usr/lib/micropython', '/mnt/c/Users/robocoder/Downloads/sp_emu/sys', '.frozen', '/mnt/c/Users/robocoder/Downloads/sp_emu/sys/bin', '.frozen', '/mnt/c/Users/robocoder/Downloads/sp_emu/sys/bin/lib', '/mnt/c/Users/robocoder/Downloads/sp_emu/sys/bin/hal']
I: /mnt/c/Users/robocoder/Downloads/sp_emu/sys/bin already in list.
Traceback (most recent call last):
    File "bin/entry.py", line 39, in <module>
    File "bin/entry.py", line 32, in boot_sys
    File "boot.py", line 3, in <module>
    File "firmware_build/boot_ldr.py", line 77, in <module>
ImportError: no module named 'panic_handler' 
```

The "missing file" is in the path /mnt/c/Users/robocoder/Downloads/sp_emu/sys/bin. I cannot figure out why it cannot be loaded as I have a dump of sys.path (above). Also, it loads the other files just fine.

Here's the offending code:
```
except Exception as ie:
	# Make sure we can read the panic handler code.
	sys.path.append("/bin")
	
	import panic_handler
	panic_handler.panic(ie, "init")
```

Chances are, I did something really stupid, but I cannot figure out what.

**Comment by dlech** [comment]
`"/bin"` is not the same as `"/mnt/c/Users/robocoder/Downloads/sp_emu/sys/bin"`, so MicroPython can't find the file because it is only looking in `/bin`/

**Reply by RSC-Games** [reply]
Yeah. I forgot to mention that I hooked ```sys.path``` calls and it would add the current working directory to them.
So the ```/bin``` would be converted to ```/mnt/c/Users/robocoder/Downloads/sp_emu/sys/bin```.

That's what all this is:
```
W: append hook.
D: fix_path: /mnt/c/Users/robocoder/Downloads/sp_emu/sys/bin
V: Path: ['/mnt/c/Users/robocoder/Downloads/sp_emu/bin', '.frozen', '/home/robocoder/.micropython/lib', '/usr/lib/micropython', '/mnt/c/Users/robocoder/Downloads/sp_emu/sys', '.frozen', '/mnt/c/Users/robocoder/Downloads/sp_emu/sys/bin', '.frozen', '/mnt/c/Users/robocoder/Downloads/sp_emu/sys/bin/lib', '/mnt/c/Users/robocoder/Downloads/sp_emu/sys/bin/hal']
I: /mnt/c/Users/robocoder/Downloads/sp_emu/sys/bin already in list.
```

Sorry about that.

**Comment by RSC-Games** [comment]
I might have an idea. I'm using WSL for this and the program by default switches path to root so that might cause a little problem.

**Comment by RSC-Games** [comment, marked as answer]
I found out the issue.
I was apparently unmounting root in the code I was trying to run but because the panic handler wasn't running I couldn't see that.
Thanks for your time though.
